### PR TITLE
fix(capacitor): update wizard

### PR DESCRIPTION
**Pull request by lucas-zimerman**
This PR updates the getting started for the Capacitor SDK, making it more dynamic and easier to use and read.

Since the SDK allows the interaction with other SDKs like Angular, React and Vue, It has minor changes when a user wants to use one of the selected sibling SDKs.

Layout was validated on a self hosted instance:
image
image
image

### CODE REVIEW

File: static/app/components/onboarding/productSelection.tsx [hunk 1]
```diff
@@ -92,6 +92,7 @@ export const platformProductAvailability = {
     ProductSolution.PERFORMANCE_MONITORING,
     ProductSolution.SESSION_REPLAY,
   ],
+  capacitor: [ProductSolution.PERFORMANCE_MONITORING, ProductSolution.SESSION_REPLAY],
```
Comment (lucas-zimerman) [inline static/app/components/onboarding/productSelection.tsx:95]:
```suggestion
  capacitor: [
    ProductSolution.PERFORMANCE_MONITORING,
    ProductSolution.SESSION_REPLAY,
  ],
```

File: static/app/gettingStartedDocs/capacitor/capacitor.tsx [hunk 2]
```diff
@@ -1,229 +1,409 @@
...
+
+export enum SiblingOption {
+  ANGULARV10 = 'angularV10',
+  ANGULARV12 = 'angularV12',
+  REACT = 'react',
+  VUE3 = 'vue3',
+  VUE2 = 'vue2',
+}
+
+type PlaformOptionKey = 'siblinOption';
```
Comment (ArthurKnaus) [inline static/app/gettingStartedDocs/capacitor/capacitor.tsx:19]: typo
```suggestion
type PlaformOptionKey = 'siblingOption';
```

File: static/app/gettingStartedDocs/capacitor/capacitor.tsx [hunk 3]
```diff
@@ -1,229 +1,409 @@
...
+  errorHandlerProviders: string;
+  sentryInitContent: string;
+  siblingOption: SiblingOption;
+  newOrg?: boolean;
+  organization?: Organization;
+  platformKey?: PlatformKey;
+  projectId?: string;
+};
+
+const IsAngular = (siblingOption: SiblingOption): boolean =>
```
Comment (ArthurKnaus) [inline static/app/gettingStartedDocs/capacitor/capacitor.tsx:31]:
```suggestion
const isAngular = (siblingOption: SiblingOption): boolean =>
```

File: static/app/gettingStartedDocs/capacitor/capacitor.tsx [hunk 4]
```diff
@@ -1,229 +1,409 @@
...
+  siblingOption === SiblingOption.ANGULARV10 ||
+  siblingOption === SiblingOption.ANGULARV12;
+
+const IsVue = (siblingOption: SiblingOption): boolean =>
```
Comment (ArthurKnaus) [inline static/app/gettingStartedDocs/capacitor/capacitor.tsx:35]:
```suggestion
const isVue = (siblingOption: SiblingOption): boolean =>
```

File: static/app/gettingStartedDocs/capacitor/capacitor.tsx [hunk 5]
```diff
@@ -1,229 +1,409 @@
...
+`;
+
+const routingInstrumentationConfig = (siblingOption: SiblingOption): string =>
+  IsVue(siblingOption)
+    ? 'routingInstrumentation: SiblingSdk.VUERouterInstrumentation(router),\n'
+    : IsAngular(siblingOption)
+    ? 'routingInstrumentation: SiblingSdk.routingInstrumentation,\n'
+    : '';
+
+function performanceIntegration(siblingOption: SiblingOption): string {
```
Comment (ArthurKnaus) [inline static/app/gettingStartedDocs/capacitor/capacitor.tsx:84]:
```suggestion
function getPerformanceIntegration(siblingOption: SiblingOption): string {
```

File: static/app/gettingStartedDocs/capacitor/capacitor.tsx [hunk 6]
```diff
@@ -1,229 +1,409 @@
...
-  dsn,
-}: Partial<Pick<ModuleProps, 'dsn'>> = {}): LayoutProps['steps'] => [
-  {
-    type: StepType.INSTALL,
-    description: t(
-      'Install the Sentry Capacitor SDK alongside the sibling Sentry Angular SDK:'
-    ),
-    configurations: [
+const platformOptions: Record<PlaformOptionKey, PlatformOption> = {
+  siblinOption: {
```
Comment (ArthurKnaus) [inline static/app/gettingStartedDocs/capacitor/capacitor.tsx:40]: ATM Angular 12+ is the selected default.
Does this make sense or should we rather select React?

File: static/app/gettingStartedDocs/capacitor/capacitor.tsx [hunk 7]
```diff
@@ -1,229 +1,409 @@
...
+new SiblingSdk.Replay(),
+`;
+
+const replayOtherConfig = `
+// Session Replay
+replaysSessionSampleRate: 0.1, // This sets the sample rate at 10%. You may want to change it to 100% while in development and then sample at a lower rate in production.
+replaysOnErrorSampleRate: 1.0, // If you're not already sampling the entire session, change the sample rate to 100% when sampling sessions where errors occur.
+`;
+
+const routingInstrumentationConfig = (siblingOption: SiblingOption): string =>
```
Comment (ArthurKnaus) [inline static/app/gettingStartedDocs/capacitor/capacitor.tsx:77]:
```suggestion
const getRoutingInstrumentationConfig = (siblingOption: SiblingOption): string =>
```

**Review by ArthurKnaus — APPROVED** [review]
Thx for the improvements! 🚀

File: static/app/gettingStartedDocs/capacitor/capacitor.tsx (hunk 6 above)
Comment (lucas-zimerman) [reply static/app/gettingStartedDocs/capacitor/capacitor.tsx]: I see no problem in selecting Angular, but if needed I can change it to React.

**Comment by lucas-zimerman** [comment]
@ArthurKnaus I have changed the getting started with the Requested changes + refactored the way it works based on the latest chances on the getting started from capacitor.

File: static/app/gettingStartedDocs/capacitor/capacitor.tsx (hunk 6 above)
Comment (ArthurKnaus) [reply static/app/gettingStartedDocs/capacitor/capacitor.tsx]: k 👍 was just wondering which of those is more common when using Capacitor.

**Review by ArthurKnaus — APPROVED** [review]
🙌 thx for adjusting

**Pull request merged**